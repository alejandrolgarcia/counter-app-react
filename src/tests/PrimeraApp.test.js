import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';

import PrimeraApp from '../PrimeraApp';

describe('Prueba en <PrimeraApp>', () => {

    // test('debe de mostrar el mensaje de Hola Mundo, soy Alejandro', () => {
    //     const saludo = 'Hola Mundo, soy Alejandro';
    //     const { getByText } = render( <PrimeraApp saludo={ saludo } /> );
    //     expect( getByText(saludo) ).toBeInTheDocument();
    // });

    test('debe de mostrar <PrimeraApp /> correctamente', () => {

        const saludo = 'Hola Mundo, soy Alejandro';
        const wrapper = shallow( <PrimeraApp saludo={ saludo } /> );

        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de mostrar el subtitulo enviado por props', () => {

        const saludo = 'Hola Mundo, soy Alejandro';
        const subTitulo = 'Yo soy un subTitulo';

        const wrapper = shallow( 
            <PrimeraApp 
                saludo={ saludo }
                subtitulo={ subTitulo }
            /> 
        );

        const textoParrafo = wrapper.find('p').text();
        // console.log(textoParrafo);

        expect( textoParrafo ).toBe( subTitulo);


    })
    


});