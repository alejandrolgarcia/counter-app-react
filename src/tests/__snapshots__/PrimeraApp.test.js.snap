// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <PrimeraApp> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    !!
    Hola Mundo, soy Alejandro
    !!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
